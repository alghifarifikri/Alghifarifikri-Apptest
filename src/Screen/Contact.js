/* eslint-disable react-native/no-inline-styles */
// import {Image} from 'native-base';
import React, {useState} from 'react';
import {
  View,
  Text,
  FlatList,
  TouchableOpacity,
  Image,
  StyleSheet,
} from 'react-native';

export default function Contact({
  loading = false,
  data = [],
  addBtn = () => {},
  onPress = () => {},
  refresh = () => {},
}) {
  return (
    <View style={styles.MainContainer}>
      <View style={{padding: 20, backgroundColor: '#953E46'}}>
        <Text
          style={{
            color: '#F4E3CF',
            textAlign: 'center',
          }}>
          All Contacts
        </Text>
      </View>
      <View
        style={{
          marginTop: 20,
          marginLeft: 10,
          marginRight: 10,
          backgroundColor: '#F4E3CF',
        }}>
        <FlatList
          style={{marginBottom: '20%'}}
          data={data}
          renderItem={({item}) => (
            <TouchableOpacity onPress={() => onPress(item)}>
              <View style={{flexDirection: 'row'}}>
                <View>
                  <Image
                    style={{width: 50, height: 50, borderRadius: 50}}
                    source={{
                      uri: item.photo,
                    }}
                  />
                </View>
                <View style={{marginLeft: 10, marginTop: 5}}>
                  <Text style={{color: '#953E46'}}>
                    {item.firstName} {item.lastName}
                  </Text>
                  <Text style={{color: '#953E46'}}>Age {item.age}</Text>
                </View>
              </View>
              <View
                style={{margin: 10, height: 1, backgroundColor: '#953E46'}}
              />
            </TouchableOpacity>
          )}
          keyExtractor={item => item.id}
          onRefresh={() => refresh()}
          refreshing={loading}
        />
      </View>
      <TouchableOpacity
        testID={'button-add'}
        activeOpacity={0.5}
        onPress={() => addBtn()}
        style={styles.TouchableOpacityStyle}>
        <Image
          source={{
            uri: 'https://reactnativecode.com/wp-content/uploads/2017/11/Floating_Button.png',
          }}
          style={styles.FloatingButtonStyle}
        />
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  MainContainer: {
    backgroundColor: '#F4E3CF',
    flex: 1,
  },

  TouchableOpacityStyle: {
    position: 'absolute',
    width: 50,
    height: 50,
    alignItems: 'center',
    justifyContent: 'center',
    right: 30,
    bottom: 30,
  },

  FloatingButtonStyle: {
    resizeMode: 'contain',
    width: 50,
    height: 50,
  },
});
