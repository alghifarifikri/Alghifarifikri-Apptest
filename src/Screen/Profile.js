/* eslint-disable react-native/no-inline-styles */
// import {Image} from 'native-base';
import React, {useState} from 'react';
import {
  View,
  Text,
  TextInput,
  TouchableOpacity,
  Image,
  StyleSheet,
} from 'react-native';

export default function Profile({
  data = {},
  loading = false,
  loadingPost = false,
  loadingDelete = false,
  show = false,
  edit = false,
  add = false,
  onPress = () => {},
  onChange = () => {},
  onDelete = () => {},
}) {
  return (
    <View style={styles.MainContainer}>
      <View style={{padding: 15, backgroundColor: '#953E46'}}>
        <View>
          <Image
            style={{
              width: 150,
              height: 150,
              marginBottom: -40,
              borderRadius: 100,
              alignSelf: 'center',
            }}
            source={{
              uri:
                data.photo !== 'N/A'
                  ? data.photo
                  : 'https://www.pngfind.com/pngs/m/610-6104451_image-placeholder-png-user-profile-placeholder-image-png.png',
            }}
          />
        </View>
      </View>
      <View
        style={{
          marginTop: 50,
          marginLeft: 10,
          marginRight: 10,
          backgroundColor: '#F4E3CF',
        }}>
        <View style={{marginTop: 20}}>
          <Text style={styles.title}>First Name</Text>
          <TextInput
            testID="input-firstname"
            onChangeText={e => onChange({firstName: e})}
            style={styles.textInput}
            value={data?.firstName}
            // editable={show ? false : true}
          />
        </View>
        <View style={{marginTop: 20}}>
          <Text style={styles.title}>Last Name</Text>
          <TextInput
            testID="input-lastname"
            onChangeText={e => onChange({lastName: e})}
            style={styles.textInput}
            value={data?.lastName}
            // editable={show ? false : true}
          />
        </View>
        <View style={{marginTop: 20}}>
          <Text style={styles.title}>Age</Text>
          <TextInput
            testID="input-age"
            onChangeText={e => onChange({age: Number(e)})}
            style={styles.textInput}
            value={data?.age?.toString()}
            // editable={show ? false : true}
          />
        </View>
      </View>
      <View style={{flexDirection: 'row'}}>
        {add ? null : (
          <View
            style={{
              backgroundColor: loadingDelete ? 'gray' : '#953E46',
              marginTop: 100,
              marginBottom: 100,
              marginLeft: 30,
              borderRadius: 20,
              width: '30%',
            }}>
            <TouchableOpacity
              testID={'button-delete'}
              disabled={loadingDelete}
              onPress={() => onDelete()}>
              <Text
                style={{
                  color: '#F4E3CF',
                  textAlign: 'center',
                  fontSize: 16,
                  padding: 10,
                }}>
                Delete
              </Text>
            </TouchableOpacity>
          </View>
        )}
        <View
          style={{
            backgroundColor: loading || loadingPost ? 'gray' : '#953E46',
            marginTop: 100,
            marginBottom: 100,
            marginLeft: 'auto',
            marginRight: 30,
            borderRadius: 20,
            width: '30%',
          }}>
          <TouchableOpacity
            disabled={add ? loadingPost : loading}
            testID={'button-save'}
            onPress={() => onPress()}>
            <Text
              style={{
                color: '#F4E3CF',
                textAlign: 'center',
                fontSize: 16,
                padding: 10,
              }}>
              Save
            </Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  MainContainer: {
    backgroundColor: '#F4E3CF',
    flex: 1,
  },

  TouchableOpacityStyle: {
    position: 'absolute',
    width: 50,
    height: 50,
    alignItems: 'center',
    justifyContent: 'center',
    right: 30,
    bottom: 30,
  },

  FloatingButtonStyle: {
    resizeMode: 'contain',
    width: 50,
    height: 50,
  },

  textInput: {
    alignSelf: 'stretch',
    padding: 2,
    marginHorizontal: 16,
    borderBottomColor: '#953E46',
    borderBottomWidth: 1,
    fontFamily: 'Poppins-Regular',
  },

  title: {
    alignSelf: 'stretch',
    marginHorizontal: 16,
    color: '#953E46',
    fontSize: 10,
  },
});
