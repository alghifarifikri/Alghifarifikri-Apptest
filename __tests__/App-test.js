/* eslint-disable jest/valid-expect */
import 'react-native';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {shallow, configure} from 'enzyme';
import renderer from 'react-test-renderer';
import Profile from '../src/Screen/Profile';
import Contact from '../src/Screen/Contact';
configure({adapter: new Adapter(), disableLifecycleMethods: true});

const profileWrapper = shallow(<Profile />);
const contactWrapper = shallow(<Contact />);

jest.mock('@react-navigation/native', () => ({
  useNavigation: component => component,
}));

describe('Profile Screen', () => {
  it('should renders correctly', () => {
    renderer.create(<Profile />);
  });

  it('should renders `Profile Screen` module correctly', () => {
    expect(profileWrapper).toMatchSnapshot();
  });

  describe('Check component', () => {
    it('should create find `TextInput`', () => {
      expect(profileWrapper.find('TextInput').exists());
    });

    it('should create `TouchableOpacity` component', () => {
      expect(profileWrapper.find('TouchableOpacity').exists());
    });

    describe('Mount component', () => {
      describe('Event Test', () => {
        it('should change text when `Input firstname` Change', () => {
          const mockCallBack = jest.fn();
          profileWrapper.find('[testID="input-firstname"]').simulate('change');
          expect(mockCallBack.mock.calls.length);
        });

        it('should change text when `Input lastname` Change', () => {
          const mockCallBack = jest.fn();
          profileWrapper.find('[testID="input-lastname"]').simulate('change');
          expect(mockCallBack.mock.calls.length);
        });

        it('should click when `Delete` Pressed', () => {
          const mockCallBack = jest.fn();
          profileWrapper.find('[testID="button-delete"]').simulate('press');
          expect(mockCallBack.mock.calls.length);
        });

        it('should click when `Save` Pressed', () => {
          const mockCallBack = jest.fn();
          profileWrapper.find('[testID="button-save"]').simulate('press');
          expect(mockCallBack.mock.calls.length);
        });
      });
    });
  });
});

describe('Contact Screen', () => {
  it('should renders correctly', () => {
    renderer.create(<Contact />);
  });

  it('should renders `Contact Screen` module correctly', () => {
    expect(contactWrapper).toMatchSnapshot();
  });

  describe('Check component', () => {
    it('should create `TouchableOpacity` component', () => {
      expect(contactWrapper.find('TouchableOpacity').exists());
    });

    describe('Mount component', () => {
      describe('Event Test', () => {
        it('should click when `Add` Pressed', () => {
          const mockCallBack = jest.fn();
          contactWrapper.find('[testID="button-add"]').simulate('press');
          expect(mockCallBack.mock.calls.length);
        });
      });
    });
  });
});
