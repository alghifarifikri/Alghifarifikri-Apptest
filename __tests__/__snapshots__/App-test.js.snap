// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Screen should renders \`Contact Screen\` module correctly 1`] = `ShallowWrapper {}`;

exports[`Profile Screen should renders \`Profile Screen\` module correctly 1`] = `ShallowWrapper {}`;
